Spatial.gd

```diff
@@ -929,7 +935,14 @@ func _on_NewIcon_pressed():
 	position.set_x(self.player_position.x)
 	position.set_y(self.player_position.y)
 	position.set_z(-self.player_position.z)
-	gen_new_icon(self.next_texture_trail, waypoint_icon, self.currently_active_category)
+	var texture_id: int = get_texture_index(self.next_texture_path)
```

Comment:
Why are we doing this texture index lookup every time we add a new icon, instead of having a cached`self.next_texture_index` variable like we have with next_texture_path

Reply:
Switching to `self.next_texture_index` variable. This variable will need to be updated on map changes but testing should that for 100 textures, the time to search was 180 microseconds.